docs/issue.adoc

```diff
@@ -0,0 +1,31 @@
+:page-title: @Issue
+:page-description: JUnit Jupiter extensions to add information about a related issue
+
+== Introduction
+
+The main reason tests are written to ensure the functionality of a requirement of to avoid technical problems.
```

Comment:
Also, I'm not quite sure what you are trying to say here. Requirements don't have functionality, I think? Maybe you meant feature?

Reply:
> Also, I'm not quite sure what you are trying to say here. Requirements don't have functionality, I think? Maybe you meant feature?

No I mean requirements. Functional requirements describe all professional functionality of an application, while non-functional requirements describe addional needs, like throughput, respone times, accessibility and so on. Requirements are definied by the customer (based on their use cases) and are the core of everything.

Features are, at least for me, addional functionalities which makes life easier or more compfortable.  Let's take the `forEach` loop as an example:

Example 1:

> Customer: I need an option to loop through a list. (Requirement)
> Engineer: Here you have a `while` loop. (Implemented requirement)

Requirement fullfilled, right? So let's provide a feature:

> Engineer: Because we are kind / you cry so much about the usability of the `while` loop, we added the `forEach` loop to make your like easier (Feature)

Example 2:

The customer definied the requirement more precice:

> Customer: I need an option to loop through a list of objects as long as there are elements and I want to have an object of the current iterated element so I can work with it directly, without accessing helper variables to access the object in e.g. an Array. (Requirement)
> Engineer: Here you have a `forEach` loop. (Implemented requirement)

Feature here? Hmm, well maybe `Stream`-API.

In my opinion the term `feature` is mostly used in fields of tools, because many people don't think of a customer relationship there.

This said the core of writing a test is to ensure a requirement or to avoid bugs, like `NullPointerException` or `IndexOutOfBoundException`.